crates/wasmparser/src/validator.rs

```diff
@@ -1244,6 +1245,11 @@ impl Validator {
                     crate::CanonicalFunction::ResourceRep { resource } => {
                         current.resource_rep(resource, types, offset)
                     }
+                    crate::CanonicalFunction::ThreadSpawn { func_ty_index } => current
+                        .thread_spawn(func_ty_index, types, offset, shared_everything_threads),
```

Comment:
In response to the borrows, oh I think I forgot but the `_` argument to this closure is `&WasmFeatures` I believe which should be suitable for passing in